### DISCUSSION TITLE
jsPsych Pavlovia plugin

### DISCUSSION THREAD

**Discussion by tpronk** [jspsych/jsPsych#1738] [Show and tell]
The Pavlovia plugin allows you to easily deploy a jsPsych task on [Pavlovia](https://pavlovia.org/). Once in a while we upgrade the plugin, inspired on things reported by our users on our [Discourse forum](https://discourse.psychopy.org/). However, might be handy to mention them over here as well, so I'm planning to post relevant updates in this thread.

Currently we're at version 2020.3, which you can find in this [this demo](https://gitlab.pavlovia.org/tpronk/jsPsych_SimpleReactionTime). It adds two features on top of the default plugin:
* Applying some processing to the collected data before submitting the data to Pavlovia. Inspired on [this post](https://discourse.psychopy.org/t/jspsych-saving-options-on-pavlovia/12880)
* Have something happen once Pavlovia confirms all the data has been stored. Inspired on [this post](https://discourse.psychopy.org/t/jspsych-experiment-token-randomly-consumed-no-data-stored/22246)

Cheers Thomas

**Comment by tpronk** [comment]
I present version 2020.4! This version adds the option to use Pavlovia configuration information in a successive trial. https://gitlab.pavlovia.org/tpronk/jsPsych_SimpleReactionTime

**Reply by iholzleitner** [reply]
Is there an update planned to make Pavlovia play nicely with jsPsych 7 by any chance?

**Reply by tpronk** [reply]
Sorry for spotting your comment so late @iholzleitner! There definitely is, though I haven't dived into the architecture of jsPsych 7 yet. Could you give me a little nudge by telling me what might need adjustment?

**Reply by jodeleeuw** [reply]
@tpronk check out [our migration guide](https://www.jspsych.org/7.1/support/migration-v7/), which includes tips for plugin development.

**Reply by tpronk** [reply]
Love the docs @jodeleeuw! I'll dive into it soon! Well not too soon; OST is just about to laumch a new release of PsychoPy/PsychoJS, so I'll take it easy on them :)

**Reply by tpronk** [reply]
@iholzleitner, I hope to have a little chat with the Pavlovia team next week. With a bit of luck the upgrade will be swift, so it might well be ready the week after that (though I can't guarantee that of course)

**Reply by tpronk** [reply]
Update! One of the Pavlovia devs would like to do the update to v7, so I'll take a step back for now. I'll give a shout when I hear of any results.

**Reply by apitiot** [reply]
Dear all,

Further to Thomas previous comments, I have now put in place a plugin for version 7.x of jsPsych.
It is available on the server at this URL: https://pavlovia.org/lib/jspsych-7-pavlovia-2022.1.1.js
Version 7.1.2 of jsPsych is also now available on the server at this URL: https://pavlovia.org/lib/vendors/jspsych-7.1.2

I have prepared an example that you will find it here: https://gitlab.pavlovia.org/apitiot/jspsych7_simple_reaction_time
It is jsPsych's simple reaction time experiment with the pavlovia 7.x plugin added on top.
You'll note that it is very similar in spirit to the previous approach: one simply needs to add the relevant scripts in the header (or in the body) and to add a pavlovia_init and pavlovia_finish at the start and at the end of the timeline, respectively.
I also took the opportunity offered by the redesign to remove the dependence on jQuery.

At this stage, the plugin is ready for you to play with. I would strongly encourage you to test your experiment thoroughly before deploying, as the plugin is very new. I'll keep testing on my end. And I would very much appreciate your feedback of course.

Cheers,


Alain

**Reply by tpronk** [reply]
I just discovered a merge request from a Pavlovia user that upgraded our pavlovia plugin on their own. While I would recommend using the one made by @apitiot, posting the merge request here for reference/comparison. https://gitlab.pavlovia.org/tpronk/jsPsych_SimpleReactionTime/merge_requests/2
